src/test/java/edu/kis/powp/jobs2d/events/SelectMouseFigureOptionListener.java

```diff
@@ -32,38 +23,32 @@ public SelectMouseFigureOptionListener(JPanel drawPanel, DriverManager driverMan
 
 	@Override
 	public void actionPerformed(ActionEvent e) {
-		setXYnull();
-		halfWidth = drawPanel.getWidth()/2;
-		halfHeight = drawPanel.getHeight()/2;
+		cleanHeadPosition();
+		int halfWindowHeight = drawPanel.getHeight()/2;
+		int halfWindowWidth = drawPanel.getWidth()/2;
 
-		List<DriverCommand> commands = new ArrayList<>();
-		DriverCommandManager manager = CommandsFeature.getDriverCommandManager();
-		manager.clearCurrentCommand();
-		manager.setCurrentCommand(commands, "Mouse Command");
 		this.drawPanel.addMouseListener(new MouseAdapter() {
 			@Override
 			public void mouseClicked(MouseEvent e) {
-				if(prevX == null || prevY == null){
-					updatePreviousMousePosition(e);
+				if(previousHeadXPosition == null || previousHeadYPosition == null){
+					updatePreviousMousePosition(e, halfWindowWidth, halfWindowHeight);
 				}
 
-				commands.add(new SetPositionCommand(prevX, prevY));
-				commands.add(new OperateToCommand(e.getX()- halfWidth, e.getY()- halfHeight));
+				driverManager.getCurrentDriver().setPosition(previousHeadXPosition, previousHeadYPosition);
+				driverManager.getCurrentDriver().operateTo(e.getX()- halfWindowWidth, e.getY()- halfWindowHeight);
```

Comment:
It would be nice to have e.g. right click only changing the position (without operateTo)